components/switch-account.js

```diff
@@ -0,0 +1,155 @@
+}
+
+export const useAccounts = () => useContext(AccountContext)
+
+const AnonAccount = ({ selected, onClick }) => {
+  const { isAnon, setIsAnon } = useAccounts()
+  const refreshMe = useMeRefresh()
+  return (
+    <div
+      className='d-flex flex-column me-2 my-1 text-center'
+    >
+      <AnonIcon
+        className='fill-muted'
+        width='135' height='135' style={{ cursor: 'pointer' }} onClick={async () => {
+          document.cookie = 'multi_auth.user-id=anonymous; Path=/; Secure'
+          // order is important to prevent flashes of no session
+          setIsAnon(true)
+          await refreshMe()
+        }}
+      />
+      <div className='fst-italic'>anonymous</div>
+      {isAnon && <div className='text-muted fst-italic'>selected</div>}
+    </div>
+  )
+}
+
+const Account = ({ account, className }) => {
+  const me = useMe()
+  const [name, setName] = useState(account.name)
+  const [src, setSrc] = useState(account.photoId || '/dorian400.jpg')
+  const refreshMe = useMeRefresh()
+  const { setIsAnon } = useAccounts()
+  useQuery(USER,
+    {
+      variables: { id: account.id },
+      onCompleted ({ user: { name, photoId } }) {
+        if (photoId) {
+          const src = `https://${process.env.NEXT_PUBLIC_MEDIA_DOMAIN}/${photoId}`
+          setSrc(src)
+        }
+        setName(name)
+      }
+    }
+  )
+  return (
+    <div
+      className='d-flex flex-column me-2 my-1 text-center'
+    >
+      <Image
+        width='135' height='135' src={src} style={{ cursor: 'pointer' }} onClick={async () => {
+          document.cookie = `multi_auth.user-id=${account.id}; Path=/; Secure`
+          await refreshMe()
+          // order is important to prevent flashes of inconsistent data in switch account dialog
+          setIsAnon(false)
+        }}
+      />
+      <Link href={`/${account.name}`}>@{name}</Link>
+      {Number(me?.id) === Number(account.id) && <div className='text-muted fst-italic'>selected</div>}
+    </div>
+  )
+}
+
+const AddAccount = () => {
+  const router = useRouter()
+  return (
+    <div className='d-flex flex-column me-2 my-1 text-center'>
+      <Image
+        width='135' height='135' src='/Portrait_Placeholder.webp' style={{ cursor: 'pointer' }} onClick={() => {
+          router.push({
+            pathname: '/login',
+            query: { callbackUrl: window.location.origin + router.asPath, multiAuth: true }
+          })
+        }}
+      />
+      <div className='fst-italic'>+ add account</div>
+    </div>
+  )
+}
+
+export default function SwitchAccountDialog () {
+  const { accounts } = useAccounts()
+  return (
+    <>
+      <div className='my-2'>
+        <div className='d-flex flex-row flex-wrap'>
+          <AnonAccount />
+          {
+            accounts.map((account) => <Account key={account.id} account={account} />)
+          }
+          <AddAccount />
+        </div>
+      </div>
+    </>
+  )
+}
```

Comment:
We probably want to reuse the [user list](https://github.com/stackernews/stacker.news/blob/master/components/user-list.js) component without stats here if possible here. Ideally we only have one view for listing users.

Reply:
Good idea! We could use the stats bar for other stuff like how many notifications an account has in the future.

Before:

![sn ekzyis com_sn_test02(iPhone SE)](https://github.com/stackernews/stacker.news/assets/27162016/44e46a45-7846-4532-a28b-0063adfdd27d)

Now:

![sn ekzyis com_(iPhone SE)](https://github.com/stackernews/stacker.news/assets/27162016/a5940a51-ac48-43bc-95cf-a326a8a5e461)